src/views/Home.jsx

```diff
@@ -45,6 +73,26 @@ export function Home({ setListToken }) {
 				Hello from the home (<code>/</code>) page!
 			</p>
 			<button onClick={createNewToken}>Create a new list</button>
+			<p>- or -</p>
+			<p>Join an existing shopping list by entering a three word token.</p>
+			<form id="join-shopping-list-form" onSubmit={joinExistingToken}>
+				<label htmlFor="join-token">Share Token</label>
+				<div>
+					<input
+						type="text"
+						name="join-token"
+						id="join-token"
+						value={joinToken}
+						placeholder="three word token"
+						onChange={(e) => setJoinToken(e.target.value)}
```

Comment:
I kind of wonder about using `setJoinToken()` in `onSubmit` rather than `onChange`? My impression here is that this approach would mean the state changes every single time the user adds a new letter or backspaces--could that cause performance issues for the app?

Reply:
I believe using `setJoinToken` inside `onSubmit` would only set the token when the user submits the form. But having it in `onChange` here allows the user to edit the token on input before submitting. But it’s always a good thing to double check with the mentors. Thanks!